src/rust/src/rustzcash.rs

```diff
@@ -587,19 +578,34 @@ pub extern "C" fn librustzcash_sprout_prove(
     vpub_old: u64,
     vpub_new: u64,
 ) {
-    // Load parameters from disk
-    let sprout_fs =
-        File::open(unsafe { &SPROUT_GROTH16_PARAMS_PATH }.as_ref().expect(
+    let params = {
+        use std::io::Read;
+
+        // Load parameters from disk
+        let sprout_fs = File::open(unsafe { &SPROUT_GROTH16_PARAMS_PATH }.as_ref().expect(
             "Parameters not loaded: SPROUT_GROTH16_PARAMS_PATH should have been initialized",
         ))
         .expect("couldn't load Sprout groth16 parameters file");
 
-    let mut sprout_fs = BufReader::with_capacity(1024 * 1024, sprout_fs);
+        let mut sprout_fs = BufReader::with_capacity(1024 * 1024, sprout_fs);
 
-    let params = Parameters::read(&mut sprout_fs, false)
-        .expect("couldn't deserialize Sprout JoinSplit parameters file");
+        let mut sprout_params_file = vec![];
+        sprout_fs
+            .read_to_end(&mut sprout_params_file)
+            .expect("couldn't read Sprout groth16 parameters file");
 
-    drop(sprout_fs);
+        let hash = blake2b_simd::Params::new()
+            .hash_length(64)
+            .hash(&sprout_params_file);
+
+        // b2sum sprout-groth16.params
+        if hash.as_bytes() != hex::decode("e9b238411bd6c0ec4791e9d04245ec350c9c5744f5610dfcce4365d5ca49dfefd5054e371842b3f88fa1b9d7e8e075249b3ebabd167fa8b0f3161292d36c180a").unwrap().as_slice() {
+            panic!("hash of Sprout groth16 parameters file is incorrect");
+        }
+
+        Parameters::read(&sprout_params_file[..], false)
+            .expect("couldn't deserialize Sprout JoinSplit parameters file")
```

Comment:
```suggestion
            .expect("Couldn't deserialize Sprout Groth16 parameters file")
```
There is no point in using `how_to_fix` here: we already read the file and checked its hash, and so whatever the problem is, it is not that the file needs to be re-downloaded.